src/seedu/addressbook/Main.java

```diff
@@ -106,12 +106,17 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        } catch  (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
+            return result;
```

Comment:
This catch block prints the stack trace. Maybe you can try to handle the exception better (Hint: We discussed some possible ways to do this in the tutorial).